### PR TITLE
Regex Completion + Async Completion = Failure to trigger on `[` in VB

### PR DESCRIPTION
**Pull request by ivanbasov**
Fix https://github.com/dotnet/roslyn/issues/33852

### CODE REVIEW

File: src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs [hunk 1]
```diff
@@ -262,6 +262,7 @@ private static bool IsAfterDot(ITextSnapshot snapshot, ITrackingSpan applicableT
             if (filterReason == CompletionFilterReason.Insertion &&
                 !IsPotentialFilterCharacter(typeChar) &&
                 !string.IsNullOrEmpty(filterText) &&
+                !string.Equals(filterText, typeChar.ToString(), StringComparison.OrdinalIgnoreCase) &&
                 !Helpers.IsFilterCharacter(bestOrFirstCompletionItem, typeChar, filterText))
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs]: this is a complex enough set of conditions to warrant some explaining comments IMO :)

**Comment by amcasey** [comment]
I'm seeing something weird where the dummy completion list (JS/TS returns some syntactic entries if semantic information is not available) pops up momentarily and then is replaced by the real list.

**Comment by amcasey** [comment]
Disable "only complete on tab/enter", type "cons" and complete "console".  Type "cons" and then rapidly type "t" " " - I usually end up with "console ", rather than "const ".  I doubt this was introduced by this change (I think I've seen @jaredpar complain of the same).

**Comment by amcasey** [comment]
Completion isn't bulletproof, but I don't think this change has regressed anything in TS.

**Comment by amcasey** [comment]
TS behavior LGTM.

**Comment by ivanbasov** [comment]
@dpoeschl ping

**Pull request merged**